crypto/s2n_pkey_evp.c

```diff
@@ -85,6 +85,10 @@ static EVP_PKEY_CTX *s2n_evp_pkey_ctx_new(EVP_PKEY *pkey, s2n_hash_algorithm has
 /* Our "digest-and-sign" EVP signing logic is intended to support FIPS 140-3.
  * FIPS 140-3 does not allow signing or verifying externally calculated digests
  * for RSA and ECDSA verify.
+ *
+ * However most FIPS 140-3 implementation raise an indicator and allow
+ * doing so. This implementation does not rely on such indicators.
+ *
```

Comment:
Even if we don't directly check indicators, we should not knowingly violate the FIPS 140-3 rules. This comment documents that we're aware that "digest-then-sign" is not supported by FIPS 140-3 modules for ECDSA verify or RSA sign/verify based on the tests defined for FIPS 140-3 modules. If we want to advertise support for Openssl + FIPS 140-3 and remove that README warning about only supporting FIPS 140-2, then we must make "digest-and-sign" work for Openssl.

Reply:
ACK!